### ISSUE TITLE
Provide support for a compatible TypeScript doc generator

### ISSUE THREAD

**Issue by alexdiliberto** [ember-learn/ember-cli-addon-docs#246]
**Feature Request**

Current doc generator support include [YUIDoc](https://github.com/ember-learn/ember-cli-addon-docs-yuidoc) and [ESDoc](https://github.com/ember-learn/ember-cli-addon-docs-esdoc)

A future TypeScript doc generator option could include either [TSDoc](https://github.com/Microsoft/tsdoc) or [TypeDoc](https://github.com/TypeStrong/typedoc)

**Comment by buschtoens** [comment]
Thanks for opening this issue. On the long run investing effort into tsdoc is probably more valuable.

**Comment by ef4** [comment]
I just converted a huge chunk of ember-animated to TypeScript only to discover this issue. 😭

**Comment by ef4** [comment]
Apparently the YUIDoc parser is sloppy enough to not care about the difference between JS and TS.  This seems to be working fine:

https://github.com/ember-learn/ember-cli-addon-docs-yuidoc/compare/master...ef4:ts

**Comment by pzuraq** [comment]
The one good thing about a comment only parser 😄 it works in any language. We can also customize it to make it less terrible (as I have done) though probably should just rewrite it at some point.

**Comment by rwwagner90** [comment]
@ef4 @pzuraq what do you guys suggest for TS? Should we recommend YUIDoc or should we try to implement something with TSDoc.

**Comment by ef4** [comment]
I agree with the comments above that tsdoc looks like a good bet. I started trying to integrate it just now, but decided I don't have time to do it, especially since the YUIDoc solution was so small.
